src/client/pages/OfferNew/Checkout/index.tsx

```diff
@@ -126,19 +164,84 @@ const Backdrop = styled('div')<Openable>`
   }};
 `
 
+const renderUpsellCard = (
+  variants: QuoteBundleVariant[],
+  selectedQuoteBundle: QuoteBundleVariant,
+  onAcceptDeal: (quoteIds: string[]) => void,
+  textKeys: Record<string, any>,
+) => {
+  const isSwedishAccidentAdded = selectedQuoteBundle.bundle.quotes.some(
+    ({ quoteDetails }) => isSwedishAccident(quoteDetails),
+  )
+  if (isSwedishAccidentAdded) {
+    return null
+  }
+
+  // Look for a bundle variant that includes Accident Insurance.
+  // In case it founds it, store the bundle variant and the accident
+  // insurance quote so we can use them to accept UpsellCard deal and
+  // get the price of Accident Insurance.
+  const remainingBundleVariants = variants.filter(
+    ({ id }) => id !== selectedQuoteBundle.id,
+  )
+  let betterDealBundleVariant: QuoteBundleVariant | null = null
+  let accidentInsuranceQuote: BundledQuote | null = null
+  for (const bundleVariant of remainingBundleVariants) {
+    const accidentInsuranceIndex = bundleVariant.bundle.quotes.findIndex(
+      ({ quoteDetails }) => isSwedishAccident(quoteDetails),
+    )
+
+    if (accidentInsuranceIndex !== -1) {
+      betterDealBundleVariant = bundleVariant
+      accidentInsuranceQuote =
+        betterDealBundleVariant.bundle.quotes[accidentInsuranceIndex]
+      break
+    }
+  }
+
+  if (betterDealBundleVariant === null || accidentInsuranceQuote === null) {
+    return null
+  }
+
+  const localizedPerMonth = textKeys.SIDEBAR_PRICE_SUFFIX_INTERVAL()
+  const { amount, currency } = accidentInsuranceQuote.price
+  const price = `${amount} ${currency}${localizedPerMonth}`
+  const betterDealQuoteIds = betterDealBundleVariant.bundle.quotes.map(
+    ({ id }) => id,
+  )
+
+  return (
+    <UpsellCard
+      title={textKeys.UPSELL_TITLE()}
+      description={textKeys.UPSELL_DESCRIPTION()}
+      price={price}
+      onAcceptDeal={() => onAcceptDeal(betterDealQuoteIds)}
+    />
+  )
+}
+
 export type CheckoutProps = {
-  offerData: OfferData
+  variants: QuoteBundleVariant[]
```

Comment:
We should probably agree a name for what we call these. Maybe `quoteBundleVariants`?

Reply:
You're right. I've taken the lazy path on this one based on another component used in that context, like [InsuranceSelector](https://github.com/HedvigInsurance/web-onboarding/blob/030c8620c3dd3b77bbbf90bae967334cb0eade5d/src/client/pages/OfferNew/index.tsx#L177). But, no excuses.